src/tests/ftest/util/general_utils.py

```diff
@@ -264,3 +269,92 @@ def check_pool_files(log, hosts, uuid):
             log.error("%s: %s not found", result[1], filename)
             status = False
     return status
+
+
+def stop_processes(hosts, pattern, verbose=True, timeout=60):
+    """Stop the processes on each hosts that match the pattern.
+
+    Args:
+        hosts (list): hosts on which to stop the processes
+        pattern (str): regular expression used to find process names to stop
+        verbose (bool, optional): display command output. Defaults to True.
+        timeout (int, optional): command timeout in seconds. Defaults to 60
+            seconds.
+
+    Returns:
+        dict: a dictionary of return codes keys and accompanying NodeSet
+            values indicating which hosts yielded the return code.
+            Return code keys:
+                0   No processes matched the criteria / No processes killed.
+                1   One or more processes matched the criteria and a kill was
+                    attempted.
+
+    """
+    result = {}
+    log = getLogger()
+    log.info("Killing any processes on %s that match: %s", hosts, pattern)
+    if hosts is not None:
+        commands = [
+            "rc=0",
+            "if pgrep --list-full {}".format(pattern),
+            "then rc=1",
+            "sudo pkill {}".format(pattern),
+            "if pgrep --list-full {}".format(pattern),
+            "then sleep 5",
+            "pkill --signal KILL {}".format(pattern),
+            "fi",
+            "fi",
+            "exit $rc",
```

Comment:
Not sure if this is the PR that was going to address the shutdown issue, but it seems like it would be better to set `rc=1` if the second pgrep finds processes running. Also, seems like the `sleep 5` should be after the first pkill in order to give things time to shutdown before moving on to more aggressive means of cleaning up.

Reply:
This PR may eventually have a role in the shutdown issue (by means of enabling clush to launch servers), but it is not intended to directly address the issue.  As mentioned in the docstring, the intent of the exit status (`rc`) is to indicate whether or not processes where detected (and then attempted to be killed).  Its current intended use is for ensuring the test starts clean by killing any leftover processes from a previous test and it is used in the `kill()` method of the `DaosAgentManager` and `DaosServerManger`.  In terms of the `sleep 5` placement, it probably should be moved as you suggested.  Given that this method is currently called as a means of killing processes that should have been killed in the previous test's tearDown() and tests will be unlikely to hit the first pkill anyway (its more of a safety net), would you be ok with fixing it in a future PR or should I resolve it here?

Reply:
If I understand you correctly, this sequence isn't currently used as part of the shutdown, and therefore making the suggested changes on this PR won't solve the shutdown issue. If that's the case, then I'm fine with getting this landed as-is and adjusting later in a PR focused on the shutdown problem.

Reply:
Yes, you understand me correctly.